### PR TITLE
upgrade azure identity to >= 1.16.1

### PR DESCRIPTION
**Pull request by YiMysty**
We've the security alert `Azure Identity Libraries and Microsoft Authentication Library Elevation of Privilege Vulnerability`

image

We cannot upgrade due to the octodns-azure dependency on 1.8.0 > azure-identity

Thus make a change here

### CODE REVIEW

File: octodns_azure/__init__.py [hunk 1]
```diff
@@ -42,7 +42,7 @@
 from octodns.record import GeoCodes, Record, Update
 
 # TODO: remove __VERSION__ with the next major version release
-__version__ = __VERSION__ = '0.0.7'
+__version__ = __VERSION__ = '0.0.8'
```
Comment (ross) [inline octodns_azure/__init__.py:45]: Please revert this change. We always do release bumps & changelog updates in a dedicated release PR. 

```suggestion
__version__ = __VERSION__ = '0.0.8'
```

File: setup.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ def version():
         'test': tests_require,
     },
     install_requires=(
-        'azure-identity>=1.7.1,<1.8.0',
+        'azure-identity>=1.16.1',
```
Comment (ross) [inline setup.py:50]: /cc @viranch who pinned this dep and the others below it in https://github.com/octodns/octodns-azure/commit/ddca49f975eff7806d7deb70898b0862d509b560

File: octodns_azure/__init__.py (hunk 1 above)
Comment (YiMysty) [reply octodns_azure/__init__.py]: reverted! :)

File: setup.py (hunk 2 above)
Comment (ross) [reply setup.py:50]: Verified the pinned set of requirements works locally so going to unblock by moving forward with this one and then aim to sort out the other clamping in the future. 

```console
(env) coho:octodns-azure ross$ pip freeze | grep azure
azure-common==1.1.28
azure-core==1.30.2
azure-identity==1.17.1
azure-mgmt-core==1.4.0
azure-mgmt-dns==8.1.0
azure-mgmt-privatedns==1.0.0
azure-mgmt-trafficmanager==1.1.0
msrestazure==0.6.4.post1
``

Comment (ross) [reply setup.py:50]: Clamping originally came about in https://github.com/octodns/octodns-azure/issues/23#issuecomment-1033444071 due to problems with traffic manager's lib. That was in 2022 though and this seems like a long time to have things clamped for. 🙃 

Realized what I tested with locally didn't include TrafficManager bits so probably didn't hit whatever the clamping was added for. Going to try and test that now.

Comment (ross) [reply setup.py:50]: OK. `dynamic` record that hit TM works, but holy 🐄  it takes forever to create even a single item TM record. Updates seem quicker.  Moving forward (again)

Comment (viranch) [reply setup.py:50]: Came here to realize this is already verified working.
